src/core/sync/event.d

```diff
@@ -0,0 +1,301 @@
+            if (m_event)
+                ResetEvent(m_event);
+        }
+        else version (Posix)
+        {
+            if (m_initalized)
+            {
+                pthread_mutex_lock(&m_mutex);
+                m_state = false;
+                pthread_mutex_unlock(&m_mutex);
+            }
+        }
+    }
+
+    /**
+     * Wait for the event to be signaled without timeout.
+     *
+     * Returns:
+     *  `true` if the event is in signaled state, `false` if the event is uninitialized or another error occured
+     */
+    bool wait()
+    {
+        version (Windows)
+        {
+            return m_event && WaitForSingleObject(m_event, INFINITE) == WAIT_OBJECT_0;
+        }
+        else version (Posix)
+        {
+            return wait(Duration.max);
+        }
+    }
+
+    /**
+     * Wait for the event to be signaled with timeout.
+     *
+     * Params:
+     *  tmout = the maximum time to wait
+     * Returns:
+     *  `true` if the event is in signaled state, `false` if the event was nonsignaled for the given time or
+     *  the event is uninitialized or another error occured
+     */
+    bool wait(Duration tmout)
+    {
+        version (Windows)
+        {
+            if (!m_event)
+                return false;
+
+            auto maxWaitMillis = dur!("msecs")(uint.max - 1);
+
+            while (tmout > maxWaitMillis)
+            {
+                auto res = WaitForSingleObject(m_event, uint.max - 1);
+                if (res != WAIT_TIMEOUT)
+                    return res == WAIT_OBJECT_0;
+                tmout -= maxWaitMillis;
+            }
+            auto ms = cast(uint)(tmout.total!"msecs");
+            return WaitForSingleObject(m_event, ms) == WAIT_OBJECT_0;
+        }
+        else version (Posix)
+        {
+            if (!m_initalized)
+                return false;
+
+            pthread_mutex_lock(&m_mutex);
+
+            int result = 0;
+            if (!m_state)
+            {
+                pthread_mutex_unlock(&m_mutex);
+
+                if (tmout == Duration.max)
+                {
+                    result = pthread_cond_wait(&m_cond, &m_mutex);
+                }
+                else
+                {
+                    import core.sync.config;
+
+                    timespec t = void;
+                    mktspec(t, tmout);
+
+                    result = pthread_cond_timedwait(&m_cond, &m_mutex, &t);
+                }
+
+                pthread_mutex_lock(&m_mutex);
+            }
+            if (result == 0 && !m_manualReset)
+                m_state = false;
+
+            pthread_mutex_unlock(&m_mutex);
+
+            return result == 0;
+        }
+    }
+
+private:
+    version (Windows)
+    {
+        HANDLE m_event;
+    }
+    else version (Posix)
+    {
+        pthread_mutex_t m_mutex;
+        pthread_cond_t m_cond;
+        bool m_initalized;
+        bool m_state;
+        bool m_manualReset;
+    }
+}
+
+// Test single-thread (non-shared) use.
```

Comment:
Maybe make this example public, s.t. there's at least one user-facing example.

Reply:
I think this example is a pretty bad one as it doesn't involve multiple threads. Maybe better to move the example from the changelog into the documentation?

Reply:
Sure. I just saw that the module is currently without any public example.